schema/src/lib.rs

```diff
@@ -262,6 +265,52 @@ macro_rules! impl_schema_int {
 
 impl_schema_int!(u128, u64, u32, u16, u8, i128, i64, i32, i16, i8);
 
+impl TypeId for NonZeroU32 {
```

Comment:
Let's hope the SDK devs know how to create one.

Reply:
the beauty of it is that they don't have to. They can continue using `Option<u32>` because both types have the same serialized representation. Niche optimization is not visible in the serialized format

Reply:
While it is effectively still `u32`, it is not redundant to let SDK devs know that some particular `u32` should not be zero. Such information is useful and might be handled depending on platform.

Reply:
Java and Kotlin have some support for contract-based programming annotations. @NotNull and @NonZero.